modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -405,6 +405,34 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
         if (group != 1)
             return Ptr<BackendNode>();
 
+        Size out;
+        int num = input_tensor.dimSize(0);
+        int in_h = input_tensor.dimSize(2);
+        int in_w = input_tensor.dimSize(3);
+        if (padMode.empty())
+        {
+            out.height = (in_h + 2 * pad.height - (dilation.height * (kernel.height - 1) + 1)) / stride.height + 1;
+            out.width = (in_w + 2 * pad.width - (dilation.width * (kernel.width - 1) + 1)) / stride.width + 1;
+        }
+        else
+        {
+            getConvPoolOutParams(Size(in_w, in_h), kernel, stride, padMode, dilation, out);
+        }
+        int64 flops = (out.area()*out_channel*num)*(CV_BIG_INT(2)*kernel.area()*in_channel+1);
+
+        // FIXME
+        //
+        // This is a workaround for GPU hang on heavy convolution workload.
+        // For the long time task, vkWaitForFences() return without error but next call on
+        // vkQueueSubmit() return -4, i.e. "VK_ERROR_DEVICE_LOST" and driver reports GPU hang.
+        //
+        // Need more investigation on root cause of GPU hang and need to optimize convolution shader
+        // to reduce process time.
+        if (flops > CV_BIG_INT(10) * 1024 * 1024 * 1024)
+        {
+            return Ptr<BackendNode>();
+        }
```

Comment:
Can we use `getFLOPS` instead using `getMemoryShapes` to determine output shapes?

Reply:
Hi @dkurt
Fixed in the new update.